components/abstractions/src/main/java/com/microsoft/kiota/authentication/BasicAccessAuthenticationProvider.java

```diff
@@ -0,0 +1,32 @@
+package com.microsoft.kiota.authentication;
+
+import com.microsoft.kiota.RequestInformation;
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+// https://en.wikipedia.org/wiki/Basic_access_authentication
+public class BasicAccessAuthenticationProvider implements AuthenticationProvider {
+
+    private final String username;
+    private final String password;
+    private final String encoded;
+
+    public BasicAccessAuthenticationProvider(String username, String password) {
+        this.username = username;
```

Comment:
check for string being null of empty

Reply:
in other parts of the codebase we are only checking for null, done like that.